RecoVertex/KinematicFitPrimitives/src/TrackKinematicStatePropagator.cc

```diff
@@ -58,6 +63,8 @@ bool TrackKinematicStatePropagator::willPropagateToTheTransversePCA(const Kinema
   // copied from below...
   FreeTrajectoryState const& fState = state.freeTrajectoryState();
   std::pair<HelixBarrelPlaneCrossingByCircle, BoundPlane::BoundPlanePointer> cros = planeCrossing(fState, point);
+  if (cros.second == nullptr)
```

Comment:
Is this return statement and the one below on L88 both supposed to be active at the same time?  Running on the 100 event test file provided by Shazhad, it seems that only the one below on L88 is executed (twice), but this one is never reached.

Reply:
this method `willPropagateToTheTransversePCA` is used to check if the propagation is possible. I can see how someone may want to call it separately from the actual propagation in `propagateToTheTransversePCACharged`.

I guess in the tested setup this part is never called.

Reply:
the `willPropagate*` is used in `KinematicParticleVertexFitter`; here the call traces go through `KinematicConstrainedVertexFitterT`.